src/additional_types.rs

```diff
@@ -60,3 +63,159 @@ impl<'a> FromSql<'a> for RustMacAddr8 {
         true
     }
 }
+
+build_additional_rust_type!(RustLine, Line);
+build_additional_rust_type!(RustPolygon, Polygon);
+
+impl<'a> FromSql<'a> for RustLine {
+    fn from_sql(
+        _ty: &Type,
+        raw: &'a [u8],
+    ) -> Result<RustLine, Box<dyn std::error::Error + Sync + Send>> {
+        if raw.len() == 4 {
+            let mut vec_raw = vec![];
+            vec_raw.extend_from_slice(raw);
+            let mut buf = vec_raw.as_slice();
+
+            let x1 = buf.read_f64::<BigEndian>()?;
+            let y1 = buf.read_f64::<BigEndian>()?;
+            let first_coord = coord!(x: x1, y: y1);
+
+            let x2 = buf.read_f64::<BigEndian>()?;
+            let y2 = buf.read_f64::<BigEndian>()?;
+            let second_coord = coord!(x: x2, y: y2);
+
+            let new_line = Line::new(first_coord, second_coord);
+            return Ok(RustLine::new(new_line));
+        }
+        Err("Cannot convert PostgreSQL LINE into rust Line".into())
+    }
+
+    fn accepts(_ty: &Type) -> bool {
+        true
+    }
+}
+
+impl<'a> FromSql<'a> for RustPolygon {
+    fn from_sql(
+        _ty: &Type,
+        raw: &'a [u8],
+    ) -> Result<RustPolygon, Box<dyn std::error::Error + Sync + Send>> {
+        if raw.len() % 2 == 0 {
+            let mut vec_raw = vec![];
+            vec_raw.extend_from_slice(raw);
+            let mut buf = vec_raw.as_slice();
+
+            let mut vec_raw_coord = vec![];
+            buf.read_f64_into::<BigEndian>(&mut vec_raw_coord);
+
+            let mut vec_coord = vec![];
+            for (x1, y1) in vec_raw_coord.into_iter().tuples() {
+                vec_coord.push(coord!(x: x1, y: y1));
+            }
+
+            let polygon_exterior = LineString::new(vec_coord);
+            let new_polygon = Polygon::new(polygon_exterior, vec![]);
+            return Ok(RustPolygon::new(new_polygon));
+        }
+        Err("Cannot convert PostgreSQL POLYGON into rust Polygon".into())
```

Comment:
For better readability - please make reverse if clause 
```rust
if raw.len() % 2 != 0 { 
    Err("Cannot convert PostgreSQL POLYGON into rust Polygon".into())
}
// Here make all convertions 
let mut vec_raw = vec![];
vec_raw.extend_from_slice(raw);
let mut buf = vec_raw.as_slice();

let mut vec_raw_coord = vec![];
buf.read_f64_into::<BigEndian>(&mut vec_raw_coord);

let mut vec_coord = vec![];
for (x1, y1) in vec_raw_coord.into_iter().tuples() {
    vec_coord.push(coord!(x: x1, y: y1));
}

let polygon_exterior = LineString::new(vec_coord);
let new_polygon = Polygon::new(polygon_exterior, vec![]);
return Ok(RustPolygon::new(new_polygon));
````

Reply:
fixed